src/classes.py

```diff
@@ -32,46 +36,77 @@ def switch_lanes(self,left_back,right_back,left_front,right_front): #params here
     #  * Take an exit
     def random_mood(self):
         return random()
-    def choose_action(self,car_env):
-        self.front, self.num_of_lanes, self.current_speed, self.speed_limit, left_back,right_back,left_front,right_front = car_env
-        
-        #check if switching lane is possible
-        switch_lane = self.switch_lanes(left_back,right_back,left_front,right_front)
-        if(self.current_speed*1 > self.front):
-            if switch_lane: 
+
+    def choose_action(self, car_env):
+        self.front, self.num_of_lanes, self.current_speed, self.speed_limit, left_back, right_back, left_front, right_front, self.frontSpeed = car_env
+
+        # check if switching lane is possible
+        switch_lane = self.switch_lanes(left_back, right_back, left_front, right_front)
+        #Variables for (de)acceleration
+        # *Delta v = Difference in speed of car infront and current car
+        # * t = reaction time
+        # * s0 = Desired minimum distance between cars
+        # * si = Distance to car in front
+        # * v = Current speed of car
+        # * ai = Max acceleration of car, taken from average acceleration in m/s^2 of cars
+        # * b = Comfortable deaccleration of car, taken from google search
+        # * v0 = desired speed of car, set as the speed limit
+        deltav = self.current_speed - self.frontSpeed
+        t = self.reaction_time
+        s0 = 3
+        si = self.front
+        v = self.current_speed
+        ai = 3.6
+        b = 3.6
+        v0 = self.speed_limit
+
+        if (self.current_speed * 1 > self.front):
+            if switch_lane:
                 action = "change_lane"
-                params = {"direction":switch_lane}
+                params = {"direction": switch_lane}
             else:
-                adjust = (self.current_speed-self.front/1) + 10 # adjust
-                if(self.current_speed - adjust) < 0: adjust = self.current_speed
+                a = ai * (1 - (v / v0) ** 4) - ai * (
+                            ((s0 + v * t) / si) + ((v * deltav) / (2 * si * math.sqrt(ai * b)))) ** 2
+
+                adjust = (self.current_speed - self.front / 1) + 10  # adjust
+                if (self.current_speed - adjust) < 0: adjust = self.current_speed
                 action = "brake"
-                params = {"value":adjust}
+                params = {"value": a}
 
-        elif(self.front > self.speed_limit + 20 and self.random_mood() > self.mood):
```

Comment:
Different condition needed. Tested with `(self.front > 20 and self.random_mood() > self.mood - 0.1)` works better
![multiple](https://user-images.githubusercontent.com/73276079/141811111-6e511f40-09c0-4990-b53b-36cc557f1d46.gif)